mirrord/cli/src/external_proxy.rs

```diff
@@ -101,6 +101,12 @@ pub async fn proxy(listen_port: u16, watch: drain::Watch) -> CliResult<()> {
         config.external_proxy.start_idle_timeout,
     )));
 
+    let mut own_agent_conn = connect_and_ping(&config, agent_connect_info.clone(), &mut analytics)
+        .await
+        .inspect_err(|_| cancellation_token.cancel())?;
```

Comment:
At this point the cancellation token still isn't used anywhere, so I don't think this line is needed, is it?

Reply:
Yep, not needed here